test/framework/src/main/java/org/opensearch/index/engine/TranslogHandler.java

```diff
@@ -95,21 +92,13 @@ public TranslogHandler(NamedXContentRegistry xContentRegistry, IndexSettings ind
     private DocumentMapperForType docMapper(String type) {
         RootObjectMapper.Builder rootBuilder = new RootObjectMapper.Builder(type);
         DocumentMapper.Builder b = new DocumentMapper.Builder(rootBuilder, mapperService);
-        return new DocumentMapperForType(b.build(mapperService), mappingUpdate);
+        return new DocumentMapperForType(b.build(mapperService), null);
     }
 
     private void applyOperation(Engine engine, Engine.Operation operation) throws IOException {
         switch (operation.operationType()) {
             case INDEX:
-                Engine.Index engineIndex = (Engine.Index) operation;
```

Comment:
I am concerned about this change: indeed `mapping.merge` does not modify the source mapping (at least, mush not), BUT removing `mapping.merge(update, MapperService.MergeReason.MAPPING_RECOVERY)` may now hide any issues which may have been surfaced during the mapping reconciliation (fe check ObjectMapper::doMerge).

Reply:
This is stale code as of #2077 since translog is no longer used in recovery

Reply:
Ah I see, could we remove it?

Reply:
Might look into doing that in another PR?

Reply:
Might look into doing that in another PR?

Reply:
👍